processing/src/main/java/org/apache/druid/query/aggregation/AggregatorFactory.java

```diff
@@ -69,6 +70,11 @@ public boolean canVectorize()
 
   public abstract Comparator getComparator();
 
+  public Comparator getComparatorForFinalOrdering(OrderByColumnSpec columnSpec)
```

Comment:
Hmm, on second look, it looks like the `getComparator()` method is for final ordering. I'm guess `getComparator()` implementation was just wrong for string first/last aggregatorFactories. Probably it could be fixed by just returning the comparator you added in `getComparator()` without adding this new method.

Reply:
> AggregatorFactory provides a unified way to get comparator (`getComparator()`), but unlike other value types, strings can be compared in several ways. it seems that `getComparator(orderingType)` method should be more suitable for string first/last agg, but it is very confusing for other types of agg.

As I mentioned above, that's why I add this new method.